starlite/handlers/base.py

```diff
@@ -85,8 +86,13 @@ def __init__(
         self.name = name
         self.opt: Dict[str, Any] = opt or {}
         self.owner: Optional[Union["Controller", "Router"]] = None
+        self._source_route_handler: Optional[Any] = None
```

Comment:
@Goldziher What should be the exact type hint here? The method that assigns value to this attribute can assign either of type `HTTPRouteHandler` or `WebSocketRouteHandler`.

Reply:
It can be `BaseRouteHandler`, but I dont understand why these changes are required.

Reply:
See `starlite.router.Router.route_handler_method_view`. Unlike `route_handler_method_map` which maps a path to alias route handlers, `route_handler_method_view` maps true route handler to its path so that decorators can resolve a path when given a route handler.